### DISCUSSION TITLE
whereHas feature — even more sugar

### DISCUSSION THREAD

**Discussion by Cellard** [laravel/framework#32825] [General]
Lets say, we have models `City` and `Street` with `hasMany` and `belongsTo` relationships.

If we need to find the city that contains the street, we write:

```
$city = City::whereHas('street', function($query) use ($street) {
   $query->whereKey($street->id);
})->first();
```

But it may be simplified that way:

```
$city = City::whereHas('street', $street)->first();
```

Or even:

```
$cities = City::whereHas('street', $collectionOfStreets);
```

Framework can do this work. 

What do you think about this feature?

(Of course, the feature is also applicable to `whereHasMorph`)

**Comment by hailwood** [comment]
It's not a bad idea, similar to something I was thinking the other day.

```php
$city->streets()->whereKey($street->id)->exists()
```

Would be cool if you could do

```php
$city->streets()->contains($street->id);
```
